tests/Doctrine/Tests/ORM/Utility/IdentifierFlattenerEnumIdTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Utility;
+
+use Doctrine\ORM\Exception\ORMException;
+use Doctrine\Tests\Models\Enums\Suit;
+use Doctrine\Tests\Models\Enums\TypedCardEnumCompositeId;
+use Doctrine\Tests\Models\Enums\TypedCardEnumId;
+use Doctrine\Tests\Models\Enums\Unit;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * Test the IdentifierFlattener utility class
+ *
+ * @requires PHP 8.1
+ * @covers \Doctrine\ORM\Utility\IdentifierFlattener
+ */
+class IdentifierFlattenerEnumIdTest extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        try {
+            $this->_schemaTool->createSchema(
+                [
+                    $this->_em->getClassMetadata(TypedCardEnumId::class),
+                    $this->_em->getClassMetadata(TypedCardEnumCompositeId::class),
+                ]
+            );
+        } catch (ORMException $e) {
+        }
+    }
+
+    /**
+     * @group utilities
+     */
+    public function testFlattenIdentifierWithEnumId(): void
+    {
+        $typedCardEnumIdEntity       = new TypedCardEnumId();
+        $typedCardEnumIdEntity->suit = Suit::Clubs;
+
+        $this->_em->persist($typedCardEnumIdEntity);
+        $this->_em->flush();
+
+        $findTypedCardEnumIdEntityNotFound = $this->_em->getRepository(TypedCardEnumId::class)->find(Suit::Diamonds);
+
+        self::assertNull($findTypedCardEnumIdEntityNotFound, 'Search by non-cached Enum ID does not work');
+
+        $findTypedCardEnumIdEntity = $this->_em->getRepository(TypedCardEnumId::class)->find(Suit::Clubs);
+
+        self::assertNotNull($findTypedCardEnumIdEntity, 'Search by Enum ID does not work');
+
+        $class = $this->_em->getClassMetadata(TypedCardEnumId::class);
+
+        $id = $class->getIdentifierValues($findTypedCardEnumIdEntity);
+
+        self::assertCount(1, $id, 'We should have 1 identifier');
+
+        self::assertEquals($id['suit'], Suit::Clubs->value);
```

Comment:
```suggestion
        self::assertEquals($id['suit'], (Suit::Clubs)->value);
```

… to avoid parse errors on PHP 7.

Reply:
There are more places like this in this file.

Reply:
if I fix this we have then PHPCS error of "Useless parenthesis". I have put annotation of @requires PHP 8.1 on top of the whole class, why is the test even loaded and run by PHPUnit on older PHP versions?

Reply:
I have tested locally with PHP 7.4 and the test is indeed skipped. Do you want me to fix it with parenthesis and get that PHPCS issue instead?

Reply:
The test has to be parsed first before PHPUnit can read the annotation. That's why we need to make sure, a PHP 7.1 parser can read the file even if the test is skipped completely.